plugins/processors/noise/noise.go

```diff
@@ -0,0 +1,156 @@
+package noise
+
+import (
+	"fmt"
+	"math"
+	"reflect"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/plugins/processors"
+	"gonum.org/v1/gonum/stat/distuv"
+)
+
+const (
+	defaultScale     = 1.0
+	defaultMin       = -1.0
+	defaultMax       = 1.0
+	defaultMu        = 0.0
+	defaultNoiseType = "laplacian"
+)
+
+const sampleConfig = `
+    ## Specified the type of the random distribution.
+    ## Can be "laplacian", "gaussian" or "uniform".
+    # type = "laplacian
+
+    ## Center of the distribution.
+    ## Only used for Laplacian and Gaussian distributions.
+    # mu = 0.0
+
+    ## Scale parameter for the Laplacian or Gaussian distribution
+    # scale = 1.0
+
+    ## Upper and lower bound of the Uniform distribution
+    # min = -1.0
+    # max = 1.0
+
+    ## Apply the noise only to numeric fields matching the filter criteria below.
+    ## Excludes takes precedence over includes.
+    # include_fields = []
+    # exclude_fields = []
+`
+
+type Noise struct {
+	Scale         float64         `toml:"scale"`
+	Min           float64         `toml:"min"`
+	Max           float64         `toml:"max"`
+	Mu            float64         `toml:"mu"`
+	IncludeFields []string        `toml:"include_fields"`
+	ExcludeFields []string        `toml:"exclude_fields"`
+	NoiseType     string          `toml:"type"`
+	Log           telegraf.Logger `toml:"-"`
+	generator     distuv.Rander
+	fieldFilter   filter.Filter
+}
+
+func (p *Noise) SampleConfig() string {
+	return sampleConfig
+}
+
+func (p *Noise) Description() string {
+	return "Adds noise to numerical fields"
+}
+
+// generates a random noise value depending on the defined probability density
+// function and adds that to the original value. If any integer overflows
+// happen during the calculation, the result is set to MaxInt or 0 (for uint)
+func (p *Noise) addNoise(value interface{}) interface{} {
+	n := p.generator.Rand()
+	switch v := value.(type) {
+	case int:
+	case int8:
+	case int16:
+	case int32:
+	case int64:
+		if v > 0 && (n > math.Nextafter(float64(math.MaxInt64), 0) || int64(n) > math.MaxInt64-v) {
+			p.Log.Debug("Int64 overflow, setting value to MaxInt64")
+			return int64(math.MaxInt64)
```

Comment:
This changes type. Not sure if that's going to be a problem. This isn't a problem for influxdb because all integers are stored as signed int64, but it may be a problem for other outputs.

Reply:
Thx, what would you propose? Copy the overflow check procedure for each ``case`` and adapt the Max/Min check values?

Reply:
I have two concerns with changing type: first, that it will affect new users of noise processors who also use outputs that pay attention to types. For them, changing the type could be a problem that prevents them from using the processor. Second, if we change types, people will get accustomed to the type changing. Then when golang generics become available and we fix the code not to change type, we'll have to accommodate users who expect the type to change.

Even though it's repetitive code and annoying, I would prefer that each type has its own case and always returns its own type. Then when golang generics are available we can simplify the code with no change to behavior.

Reply:
I agree with @reimda. Not changing the type is the right thing to do. Sorry @wizarq for not seeing it earlier!

Reply:
Thx for the feedback. You're right, the use of generics would simplify that quite a bit. For now, overflow checks are added for each type. Should we check for that in the tests as well? 
I tried the same approach for the ``(u)int8/16/32`` tests, but all values turn up to be "cast" to 64-bit...? The following test metric is handled in the ``case uint64:`` section:
```golang
testutil.MustMetric("overflow_uint32",
	map[string]string{},
	map[string]interface{}{"value": uint32(math.MaxUint32)}, 
	time.Unix(0, 0),
),
```

Reply:
@wizarq I would prefer we add a test to at least check the data types at least once each. This will be helpful for later if you ever swing back and update this to use generics. Once done, I think this is ready to merge!

Thanks!

Reply:
@powersj I agree. However I stumbled upon a problem using the table-based approach using the ``testutil.MustMetric()`` function. Apperantly the type will be cast to ``(u)int64`` and therefor always a `(u)int64` type is passed to the  ``plugin.Apply(tt.input...)`` function. 
I think this is the result of the ``convertField(v interface{})`` function [here](
https://github.com/influxdata/telegraf/blob/master/metric/metric.go#L274). Is there any reason numeric types get cast to 64 bit? 
How should we handle that?

Reply:
Yeah it seems like telegraf will _always_ convert the fields to the 64-bit types. So processors can never see any metric that is _not_ 64-bit. @powersj, @reimda Is this intended or a bug in telegraf?

Reply:
@srebhan It's not how I thought it worked so it must be a bug :) Actually after digging through the codebase history a little, it looks like it's been this way for many years. In that case, @wizarq it's safe to just handle int64 and uint64 and remove the other integer types from the switch.

Reply:
ok, thanks for the feedback. I reversed the changes so all types fall back to `(u)int64``.